R/pkg/R/LM.R

```diff
@@ -112,40 +115,31 @@ predict.sparkLM <- function(model, newData) {
 #' > Model:
 #' > y ~ x1 + x2 + x3
 #' }
-summaryObj.sparkLM <- function(model) {
-  if (class(model) != "sparkLM") {
+summary.sparkLM <- function(object, ...) {
+  if (class(object) != "sparkLM") {
     stop("model must be a sparkLM model object.")
   }
   rawSummary <- SparkR:::callJStatic("com.Alteryx.sparkGLM.LM",
                                      "summaryArray",
                                      model$jobj)
-  summaryOut <- list()
-  summaryOut$model <- rawSummary[[1]]
-  summaryOut$coefficients <- rawSummary[[2]]
-  summaryOut$RSE <- rawSummary[[3]]
-  summaryOut$R2 <- rawSummary[[4]]
-  summaryOut$Fstat <- rawSummary[[5]]
-  summaryOut$printSummary <- function() {
-    cat("\nModel:\n")
-    cat(summaryOut$model)
-    cat("\n")
-    cat("\nCoefficients:\n")
-    cat(summaryOut$coefficients, "\n")
-    cat("\n")
-    for(i in c("RSE", "R2", "Fstat")) {
-      cat(summaryOut[[i]], "\n")
-    }
-  }
-  summaryOut
+  names(rawSummary) <- c('model', 'coefficients', 'RSE', 'R2', 'Fstat')
+  class(rawSummary) <- 'summary.sparkLM'
+  return(rawSummary)
```

Comment:
It does not make a difference since R functions always return the last value. I try to be explicit wherever I can. Another way to simplify this is to replace the last three lines with a single call to `structure`

```
structure(rawSummary, 
  names =  c('model', 'coefficients', 'RSE', 'R2', 'Fstat'),
  class = 'summary.sparkLM'
)
```